src/Gui/ViewProviderCoordinateSystem.cpp

```diff
@@ -207,21 +220,24 @@ void ViewProviderCoordinateSystem::resetTemporarySize()
     });
 }
 
-void ViewProviderCoordinateSystem::updateData(const App::Property* prop) {
-    auto* jcs = dynamic_cast<App::LocalCoordinateSystem*>(getObject());
-    if(jcs) {
-        if (prop == &jcs->Placement) {
-            // Update position
-        }
-    }
+void ViewProviderCoordinateSystem::updateData(const App::Property* prop)
+{
+    //if (auto* lcs = getObject<App::LocalCoordinateSystem>()) {
+    //    if (prop == &lcs->Placement) {
+    //        // Update position
+    //    }
+    //}
     ViewProviderDocumentObject::updateData(prop);
 }
 
-bool ViewProviderCoordinateSystem::onDelete(const std::vector<std::string> &) {
-    auto lcs = getObject<App::LocalCoordinateSystem>();
+bool ViewProviderCoordinateSystem::onDelete(const std::vector<std::string> &)
+{
+    auto* lcs = getObject<App::LocalCoordinateSystem>();
+    if (!lcs) {
+        return false;
+    }
 
-    auto origin = dynamic_cast<App::Origin*>(lcs);
-    if (origin && !origin->getInList().empty()) {
+    if (lcs && !lcs->getInList().empty()) {
```

Comment:
Here we actually need to keep App::Origin. Because we do not allow deletion ONLY of origin objects. Not of normal LCS.

Reply:
Ah crap, makes sense